packages/daemon/src/context/FeedsHandler.js

```diff
@@ -123,23 +151,68 @@ export class ContactsFeedsHandler extends FeedsHandler {
         })
       },
       error: err => {
-        this._context.log('contact feed subscription error', err)
+        this._context.log('peer feed subscription error', err)
       },
       complete: () => {
-        delete this._subscriptions[contact.localID]
+        delete this._subscriptions[peerSubRef]
       },
     })
 
-    // Delete subscription if contact gets deleted
-    this._context
-      .pipe(
-        filter((e: ContextEvent) => {
-          return e.type === 'contact_deleted' && e.contactID === contact.localID
-        }),
+    return true
+  }
+
+  _subscribeContactFeed(userID: string, contact: Contact): boolean {
+    const contactSubRef = contactSubscriptionRef(contact)
+    if (this._subscriptions[contactSubRef] != null) {
+      return false
+    }
+
+    const user = this._context.openVault.identities.getOwnUser(userID)
+    const peer = this._context.openVault.identities.getPeerUser(contact.peerID)
+    if (user == null || peer == null) {
+      this._context.log(
+        `No ${user == null ? 'user' : 'peer'} found for contact`,
+        contact,
       )
-      .subscribe(() => {
-        this._subscriptions[contact.localID].unsubscribe()
+      return false
+    }
+
+    const topic = getFeedTopic({ name: user.base64PublicKey() })
+    this._subscriptions[contactSubRef] = this._context.io.bzz
+      .pollFeedValue(
+        peer.firstContactAddress,
+        {
+          mode: 'content-response',
+          whenEmpty: 'ignore',
+          contentChangedOnly: true,
+          immediate: true,
+          interval: 5 * MINUTE,
+        },
+        { topic },
```

Comment:
Not an issue but rather than calling `getFeedTopic()` manually, you could set `{ name: user.base64PublicKey() }` here

Reply:
The former will truncate a `name` over 32 characters, whereas the latter passes the `name` as a query param to swarm which will return 400 with a long `name`